src/System.Data.Common/src/System/Data/Common/DbDataReaderExtension.cs

```diff
@@ -0,0 +1,23 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Data.Common
+{
+    public static class DbDataReaderExtension
+    {
+        public static System.Collections.ObjectModel.ReadOnlyCollection<DbColumn> GetColumnSchema(this DbDataReader reader)
+        {
+            if (reader.ProvidesSchema())
+            {
+                return ((IDbColumnSchemaGenerator)reader).GetColumnSchema();
+            }
```

Comment:
This could also be implemented without using the helper method to avoid the additional cast...

```csharp
var generator = reader as IDbColumnSchemaGenerator;
if (generator != null)
{
    return generator.GetColumnSchema();
}
```

(Though, I would use `provider` over `generator` to match the type if it is renamed).